src/core/ddsi/src/q_entity.c

```diff
@@ -4005,6 +4005,10 @@ static dds_return_t new_writer_guid (struct writer **wr_out, const struct ddsi_g
       }
     }
   }
+  else
```

Comment:
This takes care of the "undef"s in the table above, except for the (A,d) case. That was always harmless, but I do think it wiser to make sure that `wr->lease` is initialised to a null pointer when not used.

Initialising `wr->lease` to a null pointer may have been the intent in `new_writer_guid_common_init`, but I have a slight preference for doing it here like you did, because it keeps setting the `lease` pointer in a single place. One could easily argue the other way as well.